daemon/cmd/daemon_main.go

```diff
@@ -1775,11 +1776,30 @@ func startDaemon(d *Daemon, restoredEndpoints *endpointRestoreState, cleaner *da
 			// This must wait for both K8s watcher caches to be synced and local endpoint restoration to be complete.
 			// Note: Synchronization of endpoints to their CEPs may not be complete at this point, but we only have to
 			// know what endpoints exist post-restoration in our endpointManager cache to perform cleanup.
-			if err := d.cleanStaleCEPs(context.Background(), d.endpointManager, params.Clientset.CiliumV2(), option.Config.EnableCiliumEndpointSlice); err != nil {
-				log.WithError(err).Error("Failed to clean up stale CEPs")
+			bo := wait.Backoff{
+				Duration: 500 * time.Millisecond,
+				Factor:   1,
+				Jitter:   0.1,
+				Steps:    5,
+				Cap:      0,
+			}
+			err := wait.ExponentialBackoffWithContext(context.Background(), bo, func(ctx context.Context) (done bool, err error) {
+				if err := d.cleanStaleCEPs(ctx, d.endpointManager, d.clientset.CiliumV2(), option.Config.EnableCiliumEndpointSlice); err != nil {
+					log.WithError(err).Error("Failed to clean up stale CEPs")
+					if resiliency.IsRetryable(err) {
+						return false, err
+					}
+					return true, err
+				}
+
+				return true, nil
+			})
+			if err != nil {
+				log.WithError(err).Error("Failed to clean up stale CEPs after retry")
```

Comment:
```suggestion
					log.WithError(err).WithField(logfields.Attempt, x).Error("Failed to clean up stale CEPs")
					if resiliency.IsRetryable(err) {
						return false, err
					}
					return true, err
				}

				return true, nil
			})
			if err != nil {
				log.WithError(err).Error("Failed to clean up stale CEPs after multiple attempts")
```

I think we can add `attempt` here in the `WithFields` for extra context.

Reply:
Good point! I'll update accordingly.